src/Microsoft.Restier.Core/Conventions/ConventionBasedEntitySetProcessor.cs

```diff
@@ -53,48 +56,130 @@ public Expression Process(QueryExpressionContext context)
             }
 
             var dataSourceStubReference = context.ModelReference as DataSourceStubModelReference;
-            if (dataSourceStubReference == null)
+            if (dataSourceStubReference != null)
+            {
+                var entitySet = dataSourceStubReference.Element as IEdmEntitySet;
+                if (entitySet == null)
+                {
+                    return null;
+                }
+
+                var collectionType = entitySet.Type as EdmCollectionType;
+                var entityType = collectionType?.ElementType.Definition as EdmEntityType;
+                if (entityType == null)
+                {
+                    return null;
+                }
+
+                return AppendOnFilterExpression(context, entityType.Name);
+            }
+
+            var propertyModelReference = context.ModelReference as PropertyModelReference;
+            if (propertyModelReference != null && propertyModelReference.Property != null)
+            {
+                // Could be a single navigation property or a collection navigation property
+                var propType = propertyModelReference.Property.Type;
+                var collectoinTypeRef = propType as IEdmCollectionTypeReference;
+                if (collectoinTypeRef != null)
+                {
+                    var collectionType = collectoinTypeRef.Definition as IEdmCollectionType;
+                    propType = collectionType.ElementType;
+                }
+
+                if (propType.TypeKind() != EdmTypeKind.Entity)
+                {
+                    return null;
+                }
+
+                var entityType = propType.Definition as EdmEntityType;
+                if (entityType == null)
+                {
+                    return null;
+                }
+
+                // In case of type inheritance, get the base type
+                var currentType = entityType;
+                while (currentType.BaseType != null)
```

Comment:
so for `Person` and its derived type `Employee`. user must write `OnFilterPerson` rather than `OnFilterEmployee`? what if he wants to filter employees based on some properties from `Employee` not `Person`?